src/coreclr/src/debug/createdump/createdump.cpp

```diff
@@ -12,13 +12,13 @@ bool g_diagnostics = false;
 bool
 CreateDump(const char* dumpPath, int pid, MINIDUMP_TYPE minidumpType)
 {
-    ReleaseHolder<DumpDataTarget> dataTarget = new DumpDataTarget(pid);
-    ReleaseHolder<CrashInfo> crashInfo = new CrashInfo(pid, dataTarget, false);
+    ReleaseHolder<CrashInfo> crashInfo = new CrashInfo(pid);
+    ReleaseHolder<DumpDataTarget> dataTarget = new DumpDataTarget(*crashInfo);
```

Comment:
Yes, you have good points and I'll clean all that up.

Reply:
I allocated and use ReleaseHolder because they are all IUnknown.  For DumpWriter, there is no reason for it to impl IUnknown.  CrashInfo/DataTarget are legitimate COM objects so I think they need to be released (with ReleaseHolder) which means they need to be allocated.